File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Adapter/Java/Main.java
public class Main {
    public static void main(String[] args) {
        USBCCharger usbcCharger = new USBCCharger();
        LightningPort adapter = new USBCToLightningAdapter(usbcCharger);

        IPhone iPhone = new IPhone(adapter);
        iPhone.charge();
    }
}

File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Adapter/Cpp/main.cpp
#include "./include/USBCCharger.h"
#include "./include/USBCToLightningAdapter.h"
#include "./include/IPhone.h"

int main() {
    USBCCharger* usbCCharger = new USBCCharger();
    LightningPort* adapter = new USBCToLightningAdapter(usbCCharger);

    IPhone* iphone = new IPhone(adapter);
    iphone->charge();

    delete iphone;
    delete adapter;
    delete usbCCharger;

    return 0;
}
